common/lib/common-utils/src/typedEventEmitter.ts

```diff
@@ -36,6 +39,58 @@ export type TypedEventTransform<TThis, TEvent> =
         // Add the default overload so this is covertable to EventEmitter regardless of environment
         TransformedEvent<TThis, EventEmitterEventType, any[]>;
 
+/** Converting from IEvent shape to EventSpec shape. Might allow for easier transition to EventSpec */
+export type ToEventArgsMapping<TEvent> =
+    //* TODO: ...Start with all 15 like IEventTransformer
```

Comment:
Todo

Reply:
You're planning on putting this EventSpec stuff in a separate PR, right?

Reply:
Nevermind, I got tripped up by some of the type names. 

nit: Rename `TEventSpec` parameter of `RequireAllStringKeys` to just `T`